app/src/androidTest/java/org/mozilla/focus/persistence/BookmarkDaoTest.java

```diff
@@ -0,0 +1,145 @@
+
+    private static final BookmarkModel BOOKMARK = new BookmarkModel("TEST_ID", "Yahoo TW", "https://tw.yahoo.com");
+    private static final BookmarkModel BOOKMARK_2 = new BookmarkModel("TEST_ID_2", "Google", "https://www.google.com");
+    private static final BookmarkModel BOOKMARK_3 = new BookmarkModel("TEST_ID_3", "GoogleV2", "https://www.google.com");
+
+    private BookmarksDatabase bookmarksDatabase;
+
+    @Before
+    public void initDb() {
+        bookmarksDatabase = Room.inMemoryDatabaseBuilder(InstrumentationRegistry.getContext(),
+                BookmarksDatabase.class)
+                // allowing main thread queries, just for testing
+                .allowMainThreadQueries()
+                .build();
+    }
+
+    @After
+    public void closeDb() {
+        bookmarksDatabase.close();
+    }
+
+    @Test
+    public void addSingleBookmark_checkBookmarkExistedInDB() throws InterruptedException {
+        // When inserting a new bookmark in the data source
+        bookmarksDatabase.bookmarkDao().addBookmarks(BOOKMARK);
+
+        // The bookmark can be retrieved
+        List<BookmarkModel> dbBookmarks = LiveDataTestUtil.getValue(bookmarksDatabase.bookmarkDao().loadBookmarks());
+        assertBookmarkEquals(BOOKMARK, dbBookmarks.get(0));
+    }
+
+    @Test
+    public void addMultipleBookmarks_checkBookmarksExistedInDB() throws InterruptedException {
+        // When inserting a bookmark list in the data source
+        bookmarksDatabase.bookmarkDao().addBookmarks(BOOKMARK, BOOKMARK_2);
+
+        // The bookmark list can be retrieved
+        List<BookmarkModel> dbBookmarks = LiveDataTestUtil.getValue(bookmarksDatabase.bookmarkDao().loadBookmarks());
+        assertEquals(2, dbBookmarks.size());
+        assertBookmarkEquals(BOOKMARK, dbBookmarks.get(0));
+        assertBookmarkEquals(BOOKMARK_2, dbBookmarks.get(1));
+    }
+
+    @Test
+    public void addMultipleBookmarks_checkSpecificBookmarkIdExistedInDB() throws InterruptedException {
+        // When inserting a bookmark list in the data source
+        bookmarksDatabase.bookmarkDao().addBookmarks(BOOKMARK, BOOKMARK_2, BOOKMARK_3);
+
+        // The bookmark with matched id can be retrieved
+        BookmarkModel dbBookmark = LiveDataTestUtil.getValue(bookmarksDatabase.bookmarkDao().getBookmarkById(BOOKMARK.getId()));
+        assertBookmarkEquals(BOOKMARK, dbBookmark);
+    }
+
+    @Test
+    public void addMultipleBookmarks_checkSpecificBookmarkUrlsExistedInDB() throws InterruptedException {
+        // When inserting a bookmark list in the data source
+        bookmarksDatabase.bookmarkDao().addBookmarks(BOOKMARK, BOOKMARK_2, BOOKMARK_3);
+
+        // The bookmark list with matched url can be retrieved
+        List<BookmarkModel> dbBookmarks = LiveDataTestUtil.getValue(bookmarksDatabase.bookmarkDao().getBookmarksByUrl(BOOKMARK_2.getUrl()));
+        assertEquals(2, dbBookmarks.size());
```

Comment:
Consider move this to another test to avoid ambiguous tests. Maybe move to addBookmarksThenDeleteByUrl_checkBookmarksDeletedInDB?